### PR TITLE
Add Breaking Changes for 2.0

### LINKED ISSUE #473: [FEATURE]: Make a breaking changes list for v2.0.0
**Is your feature request related to a problem?**
Make a new breaking changes page for v2.0.0 features. Reference https://github.com/opensearch-project/OpenSearch/issues/2480 for context.

See also breaking changes for 2.0 issue: https://github.com/opensearch-project/opensearch-clients/issues/12

**What solution would you like?**

**What alternatives have you considered?**

**Do you have any additional context?**

### PR DESCRIPTION
**Pull request by Naarcha-AWS**
Fixes #473 

Signed-off-by: Naarcha-AWS <naarcha@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by alicejw-aws — APPROVED** [review]
LGTM

File: breaking-changes.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+If you are still using the outdated terms in the context of the security APIs or for Node management, your calls and automation will continue to work until the terms are removed later in 2022. 
+
+### Deprecate Compatibility Override
+
+The override main response setting `compatibility.override_main_response_version` is deprecated from OpenSearch version 1.x and removed from OpenSearch 2.0.0. This setting is no longer supported for compatibility with legacy clients.
+
+### Add OpenSearch Notifications Plugin
+
+In OpenSearch 2.0, the Altering plugin is now integrated with the new OpenSearch Notifications plugin. To continue using the notification action in the Alerting plugin, you need to install the backend plugins `notifications-core` and `notifications`. You can also install the Notifications Dashboards plugin to manage Notification channels via OpenSearch Dashboards. For more information, see [Questions about destinations]({{site.url}}{{site.baseurl}}/monitoring-plugins/alerting/monitors#questions-about-destinations) on the Monitors page.
```
Comment (hdhalter) [inline breaking-changes.md:30]: Notifications Dashboards plugin --> Notifications plugin (?)
Altering --> Alerting 
via --> in

File: breaking-changes.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+### Removal of outdated nomenclature 
+
+In order for OpenSearch to include more inclusive naming conventions, we've replaced the following terms in our code with a more inclusive term:
+
+- "White list" is now "Allow list".
+- "Black list is now "White list".
+- "Master" is now "Cluster Manager".
+
+If you are still using the outdated terms in the context of the security APIs or for Node management, your calls and automation will continue to work until the terms are removed later in 2022.
```
Comment (keithhc2) [inline breaking-changes.md:22]: Should node be capitalized?

File: breaking-changes.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+In order for OpenSearch to include more inclusive naming conventions, we've replaced the following terms in our code with a more inclusive term:
+
+- "White list" is now "Allow list".
+- "Black list is now "White list".
+- "Master" is now "Cluster Manager".
+
+If you are still using the outdated terms in the context of the security APIs or for Node management, your calls and automation will continue to work until the terms are removed later in 2022. 
+
+### Deprecate Compatibility Override
```
Comment (natebower) [inline breaking-changes.md:24]: Sentence case?

File: breaking-changes.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+The override main response setting `compatibility.override_main_response_version` is deprecated from OpenSearch version 1.x and removed from OpenSearch 2.0.0. This setting is no longer supported for compatibility with legacy clients.
+
+### Add OpenSearch Notifications Plugin
```
Comment (natebower) [inline breaking-changes.md]: Sentence case (lowercase "plugin")?

File: breaking-changes.md (hunk 1 above)
Comment (natebower) [inline breaking-changes.md]: Third sentence: Confirm that "Notification" in "Notification channels" is a proper noun and should be capitalized.

**Pull request merged**